packages/graph/hash_graph/lib/graph/src/shared/subgraph/edges/mod.rs

```diff
@@ -1,175 +1,130 @@
+                },
+            )),
+            knowledge_graph: crate::api::utoipa::subgraph::KnowledgeGraphRootedEdges(
+                self.knowledge_graph
+                    .iter()
+                    .fold(HashMap::new(), |mut map, (id, edges)| {
+                        let edges = edges.iter().collect();
+                        match map.entry(id.base_id()) {
+                            Entry::Occupied(mut entry) => {
+                                entry.get_mut().insert(id.version(), edges);
+                            }
+                            Entry::Vacant(entry) => {
+                                entry.insert(HashMap::from([(id.version(), edges)]));
+                            }
+                        }
+                        map
+                    }),
+            ),
+        }
+        .serialize(serializer)
     }
 }
 
-#[derive(Default, Debug, Serialize)]
-pub struct Edges {
-    #[serde(flatten)]
-    ontology: OntologyRootedEdges,
-    #[serde(flatten)]
-    knowledge_graph: KnowledgeGraphRootedEdges,
+pub enum Edge {
+    Ontology {
+        edition_id: OntologyTypeEditionId,
+        edge: OntologyOutwardEdges,
+    },
+    KnowledgeGraph {
+        edition_id: EntityEditionId,
+        edge: KnowledgeGraphOutwardEdges,
+    },
 }
 
 impl Edges {
     #[must_use]
     pub fn new() -> Self {
-        Self {
-            ontology: OntologyRootedEdges(HashMap::new()),
-            knowledge_graph: KnowledgeGraphRootedEdges(HashMap::new()),
-        }
+        Self::default()
     }
 
-    /// Inserts an edge identified by `identifier` to the edge set.
+    /// Inserts an edge to the edge set.
     ///
     /// Returns whether the value was newly inserted. That is:
     ///
     /// - If the set did not previously contain this value, `true` is returned.
     /// - If the set already contained this value, `false` is returned.
-    ///
-    /// # Panics
-    ///
-    /// - if the `identifier` and `outward_edge` parameters are incompatible
-    pub fn insert(&mut self, identifier: GraphElementEditionId, outward_edge: OutwardEdge) -> bool {
-        match (identifier, outward_edge) {
-            (
-                GraphElementEditionId::Ontology(ontology_edition_id),
-                OutwardEdge::Ontology(outward_edge),
-            ) => {
-                let map = self
-                    .ontology
-                    .0
-                    .entry(ontology_edition_id.base_id().clone())
-                    .or_default();
-
-                match map.entry(ontology_edition_id.version()) {
-                    Entry::Occupied(entry) => entry.into_mut().insert(outward_edge),
-                    Entry::Vacant(entry) => {
-                        entry.insert(HashSet::from([outward_edge]));
-                        true
-                    }
+    pub fn insert(&mut self, edge: Edge) -> bool {
+        match edge {
+            Edge::Ontology { edition_id, edge } => match self.ontology.entry(edition_id) {
```

Comment:
Here we wouldn't have misleading shadowing of edge for example (if we renamed `edge` to `outward_edge`)

Reply:
Fixed in aeecf4a9275ce0d97cdf3f7071711aaa13b4ed4f